lib/tasks/plugin_assets.rake

```diff
@@ -34,61 +34,55 @@ task 'plugin:assets:precompile', :engine do |t, args|
-    _ = ActionView::Base
+  module Foreman
+    class PluginAssetsTask < Sprockets::Rails::Task
 
-    app = Rails.application
-    assets = app.config.assets
-    env = app.assets
-    target = File.join(@engine_root, 'public', 'assets')
+      attr_accessor :engine
 
-    assets.digests        = {}
-    assets.manifest       = File.join(target, @engine.engine_name)
-    assets.compile        = SETTINGS[@engine.engine_name.to_sym][:assets][:compile] || assets.compile
-    assets.compress       = SETTINGS[@engine.engine_name.to_sym][:assets][:compress] || assets.compress
-    assets.digest         = args.fetch(:digest, true)
-    assets.js_compressor  = SETTINGS[@engine.engine_name.to_sym][:assets][:js_compressor]
-
-    precompile = SETTINGS[@engine.engine_name.to_sym][:assets][:precompile]
-    precompile = fix_indexes(precompile)
+      def initialize(engine_name)
+        @engine = "#{engine_name.camelize}::Engine".constantize
+        Rails.application.config.assets.precompile = SETTINGS[@engine.engine_name.to_sym][:assets][:precompile]
+        super(Rails.application)
+      end
 
-    Sprockets::Bootstrap.new(Rails.application).run
-    compiler = Sprockets::StaticCompiler.new(env,
-                                             target,
-                                             precompile,
-                                             :manifest_path => assets.manifest,
-                                             :digest => assets.digest,
-                                             :manifest => true)
-    compiler.compile
-  end
+      def compile(digest = true)
+        with_logger do
+          manifest.compile(assets)
+        end
+      end
 
-  # Used to add index manifest files to the paths for
-  # proper resolution and addition when running Rails 3.2.8
-  # in the SCL
-  def fix_indexes(precompile)
-    if Rails.version == '3.2.8'
-      precompile.each do |asset|
-        if File.basename(asset)[/[^\.]+/, 0] == 'index'
-          asset.sub!(/\/index\./, '.')
-          precompile << asset
+      def environment
+        env = Rails.application.assets
+        Rails.application.config.assets.paths.each do |path|
+          env.append_path path
         end
+        env
+      end
+
+      def output
+        File.join(@engine.root, 'public', 'assets')
       end
-    end
 
-    precompile
+      def manifest_path
+        File.join(output, @engine.engine_name, "#{@engine.engine_name}.json")
+      end
+
+      def manifest
+        Sprockets::Manifest.new(index, output, manifest_path)
+      end
+
+    end
   end
 
   if args[:engine]
     # Partially load the Rails environment to avoid
     # the need of a database being setup
     Rails.application.initialize!(:assets)
-
-    @engine = "#{args[:engine].camelize}::Engine".constantize
-    @engine_root = @engine.root
-
-    compile_assets(:digest => false)
-    compile_assets
+    task = Foreman::PluginAssetsTask.new(args[:engine])
+    #task.compile
+    task.compile(false)
```

Comment:
What's the true/false controlling exactly - whether to produce the manifest and digests?